app/utilities/supplementary_data_parser.py

```diff
@@ -0,0 +1,97 @@
+from typing import Mapping
+
+from marshmallow import (
+    EXCLUDE,
+    INCLUDE,
+    Schema,
+    ValidationError,
+    fields,
+    validate,
+    validates_schema,
+)
+
+from app.authentication.auth_payload_versions import SupplementaryDataSchemaVersion
+from app.utilities.metadata_parser_v2 import VALIDATORS, StripWhitespaceMixin
+
+
+class ItemsSchema(Schema):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+
+
+class ItemsData(Schema, StripWhitespaceMixin):
+    pass
+
+
+class SupplementaryData(Schema, StripWhitespaceMixin):
+    identifier = VALIDATORS["string"](validate=validate.Length(min=1))
+    schema_version = VALIDATORS["string"](
+        validate=validate.OneOf([SupplementaryDataSchemaVersion.V1.value])
+    )
+    items = fields.Nested(ItemsData, required=False, unknown=EXCLUDE)
+
+    @validates_schema()
+    def validate_unit_id(self, data, **kwargs):
+        # pylint: disable=no-self-use, unused-argument
+        if (
+            data
+            and data.get("identifier")
+            and data["identifier"] != self.context["unit_id"]
+        ):
+            raise ValidationError(
+                "Supplementary data did not return the specified Unit ID"
+            )
+
+
+class SupplementaryDataMetadataSchema(Schema, StripWhitespaceMixin):
+    dataset_id = VALIDATORS["uuid"]()
+    survey_id = VALIDATORS["string"](validate=validate.Length(min=1))
+    data = fields.Nested(
+        SupplementaryData,
+        required=True,
+        unknown=EXCLUDE,
+        validate=validate.Length(min=1),
+    )
+
+    @validates_schema()
+    def validate_dataset_id(self, data, **kwargs):
```

Comment:
The name needs updating. it checks more than data set.

Reply:
Updated 👍